<!doctype html>
<html class="no-js" lang="en" data-content_root="">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" /><link rel="next" title="Team" href="about/team.html" /><link rel="prev" title="Level 3 Annotation" href="annotation/Level3_anno_epi.html" />

    <!-- Generated with Sphinx 7.1.2 and Furo 2024.01.29 -->
        <title>References - scAtlas 0.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?v=135e06be" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css?v=0a3b3ea7" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?v=36a5483c" />
    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" />
    <link rel="stylesheet" type="text/css" href="_static/css/override.css" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #003262;
  --color-brand-content: #003262;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #272822;
  --color-code-foreground: #f8f8f2;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #272822;
  --color-code-foreground: #f8f8f2;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">scAtlas 0.0.1 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon no-toc" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo only-light" src="_static/tooth_log.png" alt="Light Logo"/>
    <img class="sidebar-logo only-dark" src="_static/tooth_log.png" alt="Dark Logo"/>
  </div>
  
  
</a><form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Introduction/introduction.html">Introduction to scAtlas</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="preprocess/index.html">Preprocess</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of Preprocess</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="preprocess/metadata_process.html">Metadata Collection</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/first_round_preprocess.html">First round preprocess</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/droplet.html">Droplet discovery</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/first_round_annotation.html">First round annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/post_annotation.html">Post annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/batch_division.html">Batch division</a></li>
<li class="toctree-l2"><a class="reference internal" href="preprocess/normalization.html">Normalization</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="pre-integration/index.html">Pre-integration</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of Pre-integration</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="pre-integration/scib-benchmark.html">Integration methods benchmarking</a></li>
<li class="toctree-l2"><a class="reference internal" href="pre-integration/hypertune.html">Hyperparameter tuning</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="integration/index.html">Integration</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle navigation of Integration</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="integration/20241024_coarse_label.html">First-level annotation</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="annotation/index.html">Annotation</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><div class="visually-hidden">Toggle navigation of Annotation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="annotation/20241025_mes_cluster.html">Step 0 : Leiden Clustering in Mesenchyme</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/step1_choose_optimal_cluster_level.html">Step 1: Choose the optimal cluster levels</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/step2_mtree.html">Step 2: MRTree</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/step3_marker_detection.html">Step 3: Marker Detection</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/step4_prune_tree.html">Step 4: Prune Tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/step5_annotation.html">Step 5: Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level1_anno.html">Level 1 Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level2_anno.html">Level 2 annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level3_anno.html">Level 3 Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level4_anno.html">Level 4 annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level5_anno.html">Level 5 annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level1_anno_epi.html">Level 1 Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level2_anno_epi.html">Level 2 Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="annotation/Level3_anno_epi.html">Level 3 Annotation</a></li>
</ul>
</li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">References</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="about/team.html">Team</a></li>
<li class="toctree-l1"><a class="reference internal" href="about/cite.html">Citing scAtlas</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing Guide</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon no-toc" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="references">
<h1>References<a class="headerlink" href="#references" title="Permalink to this heading">#</a></h1>
<div class="docutils container" id="id1">
<div role="list" class="citation-list">
<div class="citation" id="id28" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>ASBoseU22<span class="fn-bracket">]</span></span>
<p>Martin Arostegui, R. Wilder Scott, Kerstin Böse, and T. Michael Underhill. Cellular taxonomy of Hic1+ mesenchymal progenitor derivatives in the limb: from embryo to adult. <em>Nature Communications</em>, 13(1):4989, August 2022. <a class="reference external" href="https://doi.org/10.1038/s41467-022-32695-1">doi:10.1038/s41467-022-32695-1</a>.</p>
</div>
<div class="citation" id="id37" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>ASGA+21<span class="fn-bracket">]</span></span>
<p>Ugur M. Ayturk, Joseph P. Scollan, Didem Goz Ayturk, Eun Sung Suh, Alexander Vesprey, Christina M. Jacobsen, Paola Divieti Pajevic, and Matthew L. Warman. Single-Cell RNA Sequencing of Calvarial and Long-Bone Endocortical Cells. <em>Journal of Bone and Mineral Research</em>, 2021.</p>
</div>
<div class="citation" id="id25" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>ButtnerMW+19<span class="fn-bracket">]</span></span>
<p>Maren Büttner, Zhichao Miao, F. Alexander Wolf, Sarah A. Teichmann, and Fabian J. Theis. A test metric for assessing single-cell RNA-seq batch correction. <em>Nature Methods</em>, 16(1):43–49, January 2019. <a class="reference external" href="https://doi.org/10.1038/s41592-018-0254-1">doi:10.1038/s41592-018-0254-1</a>.</p>
</div>
<div class="citation" id="id31" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>CJY+23<span class="fn-bracket">]</span></span>
<p>Jia Cao, Ling Jin, Zi-Qi Yan, Xiao-Kai Wang, You-You Li, Zun Wang, Yi-Wei Liu, Hong-Ming Li, Zhe Guan, Ze-Hui He, Jiang-Shan Gong, Jiang-Hua Liu, Hao Yin, Yi-Juan Tan, Chun-Gu Hong, Shi-Kai Feng, Yan Zhang, Yi-Yi Wang, Lu-Yue Qi, Chun-Yuan Chen, Zheng-Zhao Liu, Zhen-Xing Wang, and Hui Xie. Reassessing endothelial-to-mesenchymal transition in mouse bone marrow: insights from lineage tracing models. <em>Nature Communications</em>, 14(1):8461, December 2023. <a class="reference external" href="https://doi.org/10.1038/s41467-023-44312-w">doi:10.1038/s41467-023-44312-w</a>.</p>
</div>
<div class="citation" id="id38" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>CSM+20<span class="fn-bracket">]</span></span>
<p>Yuta Chiba, Kan Saito, Daniel Martin, Erich T. Boger, Craig Rhodes, Keigo Yoshizaki, Takashi Nakamura, Aya Yamada, Robert J. Morell, Yoshihiko Yamada, and Satoshi Fukumoto. Single-Cell RNA-Sequencing From Mouse Incisor Reveals Dental Epithelial Cell-Type Specific Genes. <em>Frontiers in Cell and Developmental Biology</em>, September 2020. <a class="reference external" href="https://doi.org/10.3389/fcell.2020.00841">doi:10.3389/fcell.2020.00841</a>.</p>
</div>
<div class="citation" id="id15" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>DDHediyehZadehM+23<span class="fn-bracket">]</span></span>
<p>Carlo De Donno, Soroor Hediyeh-Zadeh, Amir Ali Moinfar, Marco Wagenstetter, Luke Zappia, Mohammad Lotfollahi, and Fabian J. Theis. Population-level integration of single-cell datasets enables multi-scale analysis across samples. <em>Nature Methods</em>, pages 1–10, October 2023. <a class="reference external" href="https://doi.org/10.1038/s41592-023-02035-2">doi:10.1038/s41592-023-02035-2</a>.</p>
</div>
<div class="citation" id="id33" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>DYM+18<span class="fn-bracket">]</span></span>
<p>Shawon Debnath, Alisha R. Yallowitz, Jason McCormick, Sarfaraz Lalani, Tuo Zhang, Ren Xu, Na Li, Yifang Liu, Yeon Suk Yang, Mark Eiseman, Jae-Hyuck Shim, Meera Hameed, John H. Healey, Mathias P. Bostrom, Dan Avi Landau, and Matthew B. Greenblatt. Discovery of a periosteal stem cell mediating intramembranous bone formation. <em>Nature</em>, 562(7725):133, September 2018. <a class="reference external" href="https://doi.org/10.1038/s41586-018-0554-8">doi:10.1038/s41586-018-0554-8</a>.</p>
</div>
<div class="citation" id="id5" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>FTK+22<span class="fn-bracket">]</span></span>
<p>Thira Faruangsaeng, Sermporn Thaweesapphitak, Chompak Khamwachirapitak, Thantrira Porntaveetus, and Vorasuk Shotelersuk. Comparative transcriptome profiles of human dental pulp stem cells from maxillary and mandibular teeth. <em>Scientific Reports</em>, 12:8860, May 2022. <a class="reference external" href="https://doi.org/10.1038/s41598-022-12867-1">doi:10.1038/s41598-022-12867-1</a>.</p>
</div>
<div class="citation" id="id26" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>FMBurstynCohenS21<span class="fn-bracket">]</span></span>
<p>R. Fresia, P. Marangoni, T. Burstyn-Cohen, and A. Sharir. From Bite to Byte: Dental Structures Resolved at a Single-Cell Resolution. <em>Journal of Dental Research</em>, 100(9):897–905, August 2021. <a class="reference external" href="https://doi.org/10.1177/00220345211001848">doi:10.1177/00220345211001848</a>.</p>
</div>
<div class="citation" id="id19" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>HJY+24<span class="fn-bracket">]</span></span>
<p>Litian Han, Yaoting Ji, Yiqian Yu, Yueqi Ni, Hao Zeng, Xiaoxin Zhang, Huan Liu, and Yufeng Zhang. Trajectory-centric framework TrajAtlas reveals multi-scale differentiation heterogeneity among cells, genes, and gene modules in osteogenesis. <em>PLOS Genetics</em>, 20(10):e1011319, October 2024. <a class="reference external" href="https://doi.org/10.1371/journal.pgen.1011319">doi:10.1371/journal.pgen.1011319</a>.</p>
</div>
<div class="citation" id="id29" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>HBH+22<span class="fn-bracket">]</span></span>
<p>Florian Hermans, Celine Bueds, Lara Hemeryck, Ivo Lambrichts, Annelies Bronckaers, and Hugo Vankelecom. Establishment of inclusive single-cell transcriptome atlases from mouse and human tooth as powerful resource for dental research. <em>Frontiers in Cell and Developmental Biology</em>, 2022.</p>
</div>
<div class="citation" id="id36" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>HDW+22<span class="fn-bracket">]</span></span>
<p>Hong Hu, Yufeng Duan, Kun Wang, Huancheng Fu, Yuansong Liao, Tianshu Wang, Ziwei Zhang, Fanchen Kang, Baiquan Zhang, Haiying Zhang, Fangjun Huo, Yike Yin, Guoqing Chen, Hongbo Hu, Haoyang Cai, Weidong Tian, and Zhonghan Li. Dental niche cells directly contribute to tooth reconstitution and morphogenesis. <em>Cell Reports</em>, 41(10):111737, December 2022. <a class="reference external" href="https://doi.org/10.1016/j.celrep.2022.111737">doi:10.1016/j.celrep.2022.111737</a>.</p>
</div>
<div class="citation" id="id2" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>JFY+22<span class="fn-bracket">]</span></span>
<p>Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. <em>Nature Communications</em>, 13(1):4803, August 2022. <a class="reference external" href="https://doi.org/10.1038/s41467-022-32490-y">doi:10.1038/s41467-022-32490-y</a>.</p>
</div>
<div class="citation" id="id8" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>KSK+20<span class="fn-bracket">]</span></span>
<p>Jan Krivanek, Ruslan A. Soldatov, Maria Eleni Kastriti, Tatiana Chontorotzea, Anna Nele Herdina, Julian Petersen, Bara Szarowska, Marie Landova, Veronika Kovar Matejova, Lydie Izakovicova Holla, Ulrike Kuchler, Ivana Vidovic Zdrilic, Anushree Vijaykumar, Anamaria Balic, Pauline Marangoni, Ophir D. Klein, Vitor C. M. Neves, Val Yianni, Paul T. Sharpe, Tibor Harkany, Brian D. Metscher, Marc Bajénoff, Mina Mina, Kaj Fried, Peter V. Kharchenko, and Igor Adameyko. Dental cell type atlas reveals stem and differentiated cell types in mouse and human teeth. <em>Nature Communications</em>, 11(1):4816, September 2020. <a class="reference external" href="https://doi.org/10.1038/s41467-020-18512-7">doi:10.1038/s41467-020-18512-7</a>.</p>
</div>
<div class="citation" id="id16" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LMW+23<span class="fn-bracket">]</span></span>
<p>Blue B. Lake, Rajasree Menon, Seth Winfree, Qiwen Hu, Ricardo Melo Ferreira, Kian Kalhor, Daria Barwinska, Edgar A. Otto, Michael Ferkowicz, Dinh Diep, Nongluk Plongthongkum, Amanda Knoten, Sarah Urata, Laura H. Mariani, Abhijit S. Naik, Sean Eddy, Bo Zhang, Yan Wu, Diane Salamon, James C. Williams, Xin Wang, Karol S. Balderrama, Paul J. Hoover, Evan Murray, Jamie L. Marshall, Teia Noel, Anitha Vijayan, Austin Hartman, Fei Chen, Sushrut S. Waikar, Sylvia E. Rosas, Francis P. Wilson, Paul M. Palevsky, Krzysztof Kiryluk, John R. Sedor, Robert D. Toto, Chirag R. Parikh, Eric H. Kim, Rahul Satija, Anna Greka, Evan Z. Macosko, Peter V. Kharchenko, Joseph P. Gaut, Jeffrey B. Hodgin, Michael T. Eadon, Pierre C. Dagher, Tarek M. El-Achkar, Kun Zhang, Matthias Kretzler, and Sanjay Jain. An atlas of healthy and injured cell states and niches in the human kidney. <em>Nature</em>, 619(7970):585–594, July 2023. <a class="reference external" href="https://doi.org/10.1038/s41586-023-05769-3">doi:10.1038/s41586-023-05769-3</a>.</p>
</div>
<div class="citation" id="id24" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LMSW22<span class="fn-bracket">]</span></span>
<p>Hui Li, Davis J. McCarthy, Heejung Shim, and Susan Wei. Trade-off between conservation of biological variation and batch effect removal in deep generative modeling for single-cell transcriptomics. <em>BMC Bioinformatics</em>, 23(1):1–22, November 2022. <a class="reference external" href="https://doi.org/10.1186/s12859-022-05003-3">doi:10.1186/s12859-022-05003-3</a>.</p>
</div>
<div class="citation" id="id34" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LZX+18<span class="fn-bracket">]</span></span>
<p>Wenjia Liu, Liqiang Zhang, Kun Xuan, Chenghu Hu, Shiyu Liu, Li Liao, Bei Li, Fang Jin, Songtao Shi, and Yan Jin. Alpl prevents bone ageing sensitivity by specifically regulating senescence and differentiation in mesenchymal stem cells. <em>Bone Research</em>, 6(1):1–15, September 2018. <a class="reference external" href="https://doi.org/10.1038/s41413-018-0029-4">doi:10.1038/s41413-018-0029-4</a>.</p>
</div>
<div class="citation" id="id17" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LHTS24<span class="fn-bracket">]</span></span>
<p>Mohammad Lotfollahi, Yuhan Hao, Fabian J. Theis, and Rahul Satija. The future of rapid and automated single-cell data analysis using reference mapping. <em>Cell</em>, 187(10):2343–2358, May 2024. <a class="reference external" href="https://doi.org/10.1016/j.cell.2024.03.009">doi:10.1016/j.cell.2024.03.009</a>.</p>
</div>
<div class="citation" id="id21" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LNL+22<span class="fn-bracket">]</span></span>
<p>Mohammad Lotfollahi, Mohsen Naghipourfar, Malte D. Luecken, Matin Khajavi, Maren Büttner, Marco Wagenstetter, Žiga Avsec, Adam Gayoso, Nir Yosef, Marta Interlandi, Sergei Rybakov, Alexander V. Misharin, and Fabian J. Theis. Mapping single-cell data to reference atlases by transfer learning. <em>Nature Biotechnology</em>, 40(1):121–130, January 2022. <a class="reference external" href="https://doi.org/10.1038/s41587-021-01001-7">doi:10.1038/s41587-021-01001-7</a>.</p>
</div>
<div class="citation" id="id20" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LButtnerC+22<span class="fn-bracket">]</span></span>
<p>Malte D. Luecken, M. Büttner, K. Chaichoompu, A. Danese, M. Interlandi, M. F. Mueller, D. C. Strobl, L. Zappia, M. Dugas, M. Colomé-Tatché, and Fabian J. Theis. Benchmarking atlas-level data integration in single-cell genomics. <em>Nature Methods</em>, 19(1):41–50, January 2022. <a class="reference external" href="https://doi.org/10.1038/s41592-021-01336-8">doi:10.1038/s41592-021-01336-8</a>.</p>
</div>
<div class="citation" id="id9" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>LWL+24<span class="fn-bracket">]</span></span>
<p>S.Y. Luo, S. Wang, Z.X. Liu, Q. Bian, and X.D. Wang. Six1 Regulates Mouse Incisor Development by Promoting Dlx1/2/5 Expression. <em>Journal of Dental Research</em>, 103(10):1017–1027, September 2024. <a class="reference external" href="https://doi.org/10.1177/00220345241256286">doi:10.1177/00220345241256286</a>.</p>
</div>
<div class="citation" id="id3" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>MO21<span class="fn-bracket">]</span></span>
<p>Toshihide Mizoguchi and Noriaki Ono. The diverse origin of bone-forming osteoblasts. <em>Journal of Bone and Mineral Research</em>, 36(8):1432–1447, 2021. <a class="reference external" href="https://doi.org/10.1002/jbmr.4410">doi:10.1002/jbmr.4410</a>.</p>
</div>
<div class="citation" id="id35" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>NCO+21<span class="fn-bracket">]</span></span>
<p>Mizuki Nagata, Angel Ka Yan Chu, Noriaki Ono, Joshua D. Welch, and Wanida Ono. Single-Cell Transcriptomic Analysis Reveals Developmental Relationships and Specific Markers of Mouse Periodontium Cellular Subsets. <em>Frontiers in dental medicine</em>, 2:679937, August 2021. <a class="reference external" href="https://doi.org/10.3389/fdmed.2021.679937">doi:10.3389/fdmed.2021.679937</a>.</p>
</div>
<div class="citation" id="id6" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>PGueguenP+17<span class="fn-bracket">]</span></span>
<p>Sophie Pantalacci, Laurent Guéguen, Coraline Petit, Anne Lambert, Renata Peterkovà, and Marie Sémon. Transcriptomic signatures shaped by cell proportions shed light on comparative developmental biology. <em>Genome Biology</em>, 18:29, February 2017. <a class="reference external" href="https://doi.org/10.1186/s13059-017-1157-7">doi:10.1186/s13059-017-1157-7</a>.</p>
</div>
<div class="citation" id="id23" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>PWE+21<span class="fn-bracket">]</span></span>
<p>Minshi Peng, Brie Wamsley, Andrew G Elkins, Daniel H Geschwind, Yuting Wei, and Kathryn Roeder. Cell type hierarchy reconstruction via reconciliation of multi-resolution cluster tree. <em>Nucleic Acids Research</em>, 49(16):e91, June 2021. <a class="reference external" href="https://doi.org/10.1093/nar/gkab481">doi:10.1093/nar/gkab481</a>.</p>
</div>
<div class="citation" id="id7" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SMT+17<span class="fn-bracket">]</span></span>
<p>Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. <em>eLife</em>, 6:e24712, May 2017. <a class="reference external" href="https://doi.org/10.7554/eLife.24712">doi:10.7554/eLife.24712</a>.</p>
</div>
<div class="citation" id="id4" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SLT+21<span class="fn-bracket">]</span></span>
<p>Hui Sophie Shu, Yiming Liam Liu, Xinyu Thomas Tang, Xinyi Shirley Zhang, Bin Zhou, Weiguo Zou, and Bo O. Zhou. Tracing the skeletal progenitor transition during postnatal bone formation. <em>Cell Stem Cell</em>, 28(12):2122–2136.e3, December 2021. <a class="reference external" href="https://doi.org/10.1016/j.stem.2021.08.010">doi:10.1016/j.stem.2021.08.010</a>.</p>
</div>
<div class="citation" id="id13" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SRamirezSuasteguiS+23<span class="fn-bracket">]</span></span>
<p>Lisa Sikkema, Ciro Ramírez-Suástegui, Daniel C. Strobl, Tessa E. Gillett, Luke Zappia, Elo Madissoon, Nikolay S. Markov, Laure-Emmanuelle Zaragosi, Yuge Ji, Meshal Ansari, Marie-Jeanne Arguel, Leonie Apperloo, Martin Banchero, Christophe Bécavin, Marijn Berg, Evgeny Chichelnitskiy, Mei-i Chung, Antoine Collin, Aurore C. A. Gay, Janine Gote-Schniering, Baharak Hooshiar Kashani, Kemal Inecik, Manu Jain, Theodore S. Kapellos, Tessa M. Kole, Sylvie Leroy, Christoph H. Mayr, Amanda J. Oliver, Michael von Papen, Lance Peter, Chase J. Taylor, Thomas Walzthoeni, Chuan Xu, Linh T. Bui, Carlo De Donno, Leander Dony, Alen Faiz, Minzhe Guo, Austin J. Gutierrez, Lukas Heumos, Ni Huang, Ignacio L. Ibarra, Nathan D. Jackson, Preetish Kadur Lakshminarasimha Murthy, Mohammad Lotfollahi, Tracy Tabib, Carlos Talavera-López, Kyle J. Travaglini, Anna Wilbrey-Clark, Kaylee B. Worlock, Masahiro Yoshida, Maarten van den Berge, Yohan Bossé, Tushar J. Desai, Oliver Eickelberg, Naftali Kaminski, Mark A. Krasnow, Robert Lafyatis, Marko Z. Nikolic, Joseph E. Powell, Jayaraj Rajagopal, Mauricio Rojas, Orit Rozenblatt-Rosen, Max A. Seibold, Dean Sheppard, Douglas P. Shepherd, Don D. Sin, Wim Timens, Alexander M. Tsankov, Jeffrey Whitsett, Yan Xu, Nicholas E. Banovich, Pascal Barbry, Thu Elizabeth Duong, Christine S. Falk, Kerstin B. Meyer, Jonathan A. Kropski, Dana Pe'er, Herbert B. Schiller, Purushothama Rao Tata, Joachim L. Schultze, Sara A. Teichmann, Alexander V. Misharin, Martijn C. Nawijn, Malte D. Luecken, and Fabian J. Theis. An integrated cell atlas of the lung in health and disease. <em>Nature Medicine</em>, 29(6):1563–1577, June 2023. <a class="reference external" href="https://doi.org/10.1038/s41591-023-02327-2">doi:10.1038/s41591-023-02327-2</a>.</p>
</div>
<div class="citation" id="id22" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>SLK+22<span class="fn-bracket">]</span></span>
<p>Lukas Steuernagel, Brian Y. H. Lam, Paul Klemm, Georgina K. C. Dowsett, Corinna A. Bauder, John A. Tadross, Tamara Sotelo Hitschfeld, Almudena del Rio Martin, Weiyi Chen, Alain J. de Solis, Henning Fenselau, Peter Davidsen, Irene Cimino, Sara N. Kohnke, Debra Rimmington, Anthony P. Coll, Andreas Beyer, Giles S. H. Yeo, and Jens C. Brüning. HypoMap—a unified single-cell gene expression atlas of the murine hypothalamus. <em>Nature Metabolism</em>, 4(10):1402–1419, October 2022. <a class="reference external" href="https://doi.org/10.1038/s42255-022-00657-y">doi:10.1038/s42255-022-00657-y</a>.</p>
</div>
<div class="citation" id="id12" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>TAC+20<span class="fn-bracket">]</span></span>
<p>Hoa Thi Nhu Tran, Kok Siong Ang, Marion Chevrier, Xiaomeng Zhang, Nicole Yee Shin Lee, Michelle Goh, and Jinmiao Chen. A benchmark of batch-effect correction methods for single-cell RNA sequencing data. <em>Genome Biology</em>, 21(1):12, January 2020. <a class="reference external" href="https://doi.org/10.1186/s13059-019-1850-9">doi:10.1186/s13059-019-1850-9</a>.</p>
</div>
<div class="citation" id="id10" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>TS04<span class="fn-bracket">]</span></span>
<p>Abigail Tucker and Paul Sharpe. The cutting-edge of mammalian development; how the embryo makes teeth. <em>Nature Reviews Genetics</em>, 5(7):499–508, July 2004. <a class="reference external" href="https://doi.org/10.1038/nrg1380">doi:10.1038/nrg1380</a>.</p>
</div>
<div class="citation" id="id27" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>WZC+22<span class="fn-bracket">]</span></span>
<p>Yaofeng Wang, Yifan Zhao, Shubin Chen, Xiaoming Chen, Yanmei Zhang, Hong Chen, Yuansong Liao, Jiashu Zhang, Di Wu, Hongxing Chu, Hongying Huang, Caixia Wu, Shijuan Huang, Huichao Xu, Bei Jia, Jie Liu, Bo Feng, Zhonghan Li, Dajiang Qin, Duanqing Pei, and Jinglei Cai. Single cell atlas of developing mouse dental germs reveals populations of CD24+ and Plac8+ odontogenic cells. <em>Science Bulletin</em>, 67(11):1154–1169, June 2022. <a class="reference external" href="https://doi.org/10.1016/j.scib.2022.03.012">doi:10.1016/j.scib.2022.03.012</a>.</p>
</div>
<div class="citation" id="id11" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>WZX+15<span class="fn-bracket">]</span></span>
<p>Wei Wu, Jian Zhou, Chong-Tao Xu, Jie Zhang, Yan-Jiao Jin, and Geng-Lin Sun. Derivation and growth characteristics of dental pulp stem cells from patients of different ages. <em>Molecular Medicine Reports</em>, 12(4):5127–5134, October 2015. <a class="reference external" href="https://doi.org/10.3892/mmr.2015.4106">doi:10.3892/mmr.2015.4106</a>.</p>
</div>
<div class="citation" id="id14" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>XLM+21<span class="fn-bracket">]</span></span>
<p>Chenling Xu, Romain Lopez, Edouard Mehlman, Jeffrey Regier, Michael I Jordan, and Nir Yosef. Probabilistic harmonization and annotation of single-cell transcriptomics data with deep generative models. <em>Molecular Systems Biology</em>, 17(1):e9620, January 2021. <a class="reference external" href="https://doi.org/10.15252/msb.20209620">doi:10.15252/msb.20209620</a>.</p>
</div>
<div class="citation" id="id32" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>YAB+20<span class="fn-bracket">]</span></span>
<p>Jing Yang, Parker Antin, Geert Berx, Cédric Blanpain, Thomas Brabletz, Marianne Bronner, Kyra Campbell, Amparo Cano, Jordi Casanova, Gerhard Christofori, Shoukat Dedhar, Rik Derynck, Heide L. Ford, Jonas Fuxe, Antonio García de Herreros, Gregory J. Goodall, Anna-Katerina Hadjantonakis, Ruby Y. J. Huang, Chaya Kalcheim, Raghu Kalluri, Yibin Kang, Yeesim Khew-Goodall, Herbert Levine, Jinsong Liu, Gregory D. Longmore, Sendurai A. Mani, Joan Massagué, Roberto Mayor, David McClay, Keith E. Mostov, Donald F. Newgreen, M. Angela Nieto, Alain Puisieux, Raymond Runyan, Pierre Savagner, Ben Stanger, Marc P. Stemmler, Yoshiko Takahashi, Masatoshi Takeichi, Eric Theveneau, Jean Paul Thiery, Erik W. Thompson, Robert A. Weinberg, Elizabeth D. Williams, Jianhua Xing, Binhua P. Zhou, and Guojun Sheng. Guidelines and definitions for research on epithelial–mesenchymal transition. <em>Nature Reviews Molecular Cell Biology</em>, 21(6):341–352, June 2020. <a class="reference external" href="https://doi.org/10.1038/s41580-020-0237-9">doi:10.1038/s41580-020-0237-9</a>.</p>
</div>
<div class="citation" id="id30" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>ZLZ+24<span class="fn-bracket">]</span></span>
<p>Yingchun Zheng, Ting Lu, Leitao Zhang, Zhongzhi Gan, Aoxi Li, Chuandong He, Fei He, Sha He, Jian Zhang, and Fu Xiong. Single-cell RNA-seq analysis of rat molars reveals cell identity and driver genes associated with dental mesenchymal cell differentiation. <em>BMC Biology</em>, 22(1):1–13, September 2024. <a class="reference external" href="https://doi.org/10.1186/s12915-024-01996-w">doi:10.1186/s12915-024-01996-w</a>.</p>
</div>
</div>
</div>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="about/team.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Team</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="annotation/Level3_anno_epi.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Level 3 Annotation</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Gilbert Han
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link fab fa-github" href="https://github.com/GilbertHan1011/toothAtlasManuscript" aria-label="GitHub"></a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer no-toc">
      
      
      
    </aside>
  </div>
</div><script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=f6245a2f"></script>
    <script src="_static/doctools.js?v=888ff710"></script>
    <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/scripts/furo.js?v=32e29ea5"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/design-tabs.js?v=36754332"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script defer="defer" src="https://unpkg.com/@popperjs/core@2"></script>
    <script defer="defer" src="https://unpkg.com/tippy.js@6"></script>
    <script defer="defer" src="_static/tippy/references.bac7a72b-0a2c-4a15-83cb-99656453bd42.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    </body>
</html>