selector_to_html = {"a[href=\"#study-name-sample-name\"]": "<h3 class=\"tippy-header\" style=\"margin-top: 0;\">Study name &amp; Sample name<a class=\"headerlink\" href=\"#study-name-sample-name\" title=\"Permalink to this heading\">#</a></h3><p>Usually, we define the study as one publication conducted by one lab. Samples in the same study are usually comparable. For example, the paper <a class=\"reference external\" href=\"https://www.nature.com/articles/s41467-022-32490-y\"><strong>Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis</strong></a> <span id=\"id1\">[<a class=\"reference internal\" href=\"../references.html#id2\" title=\"Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. Nature Communications, 13(1):4803, August 2022. doi:10.1038/s41467-022-32490-y.\">Jing <em>et al.</em>, 2022</a>]</span> conducted by Yang Chai lab contains five samples ranging from E13.5 to PN7.5 in mouse embryo development. In my convention, I use characters \u201cuniqueID+author\u201d to name the study and use \u201cuniqueID+sampleName\u201d to name the sample. In this example, the study name is <code class=\"docutils literal notranslate\"><span class=\"pre\">ToothNc_Junjun</span></code> and the sample name is <code class=\"docutils literal notranslate\"><span class=\"pre\">ToothNc_Junjun_E13.5</span></code>. The most important thing is the study name should be easy to remember and unique.</p>", "a[href=\"#metadata-collection\"]": "<h1 class=\"tippy-header\" style=\"margin-top: 0;\">Metadata Collection<a class=\"headerlink\" href=\"#metadata-collection\" title=\"Permalink to this heading\">#</a></h1><h2>Introduction<a class=\"headerlink\" href=\"#introduction\" title=\"Permalink to this heading\">#</a></h2><p>When you\u2019re working on large-scale single-cell datasets, you should manually collect metadata for each study. Here are some benefits:</p>", "a[href=\"#tissue-information\"]": "<h3 class=\"tippy-header\" style=\"margin-top: 0;\">Tissue information<a class=\"headerlink\" href=\"#tissue-information\" title=\"Permalink to this heading\">#</a></h3><p>This is the most tricky part, and it is highly dependent on your purpose. For example, when I build the <a class=\"reference external\" href=\"https://trajatlas.readthedocs.io/en/stable/introduction/Differentiation_Atlas.html\">Differentiation Atlas</a> focused on bone development, I found it\u2019s hard to use one single tissue information to summarize the tissue information. So I use multi-level tissue information for description. Some reviews <span id=\"id2\">[<a class=\"reference internal\" href=\"../references.html#id3\" title=\"Toshihide Mizoguchi and Noriaki Ono. The diverse origin of bone-forming osteoblasts. Journal of Bone and Mineral Research, 36(8):1432\u20131447, 2021. doi:10.1002/jbmr.4410.\">Mizoguchi and Ono, 2021</a>]</span> suggest there are three main organs that generate the osteoblasts: Head, Limb bud, and Long bone, and the osteogenic pathways are significantly different across the organs. So I used <strong>tissue origin</strong> which contains these three categories to describe the main organ sources. And in a finer resolution, we can see that the organ can be divided into the <strong>histological</strong> and <strong>tissue location</strong> subtypes <span id=\"id3\">[<a class=\"reference internal\" href=\"../references.html#id4\" title=\"Hui Sophie Shu, Yiming Liam Liu, Xinyu Thomas Tang, Xinyi Shirley Zhang, Bin Zhou, Weiguo Zou, and Bo O. Zhou. Tracing the skeletal progenitor transition during postnatal bone formation. Cell Stem Cell, 28(12):2122-2136.e3, December 2021. doi:10.1016/j.stem.2021.08.010.\">Shu <em>et al.</em>, 2021</a>]</span>. For example, the long bone can be divided into metaphysis, diaphysis, and epiphysis, which are histologically different. And it can also be divided into tibia, femur, which is location different. Therefore, I finally form three-level tissue metadata.\n<img alt=\"png\" src=\"../_images/meta_tissue.png\"/></p><p>And in tooth development, the situation is completely different. First, the tooth from <strong>mandibular or maxillary</strong> may have great shape and transcriptional differences <span id=\"id4\">[<a class=\"reference internal\" href=\"../references.html#id5\" title=\"Thira Faruangsaeng, Sermporn Thaweesapphitak, Chompak Khamwachirapitak, Thantrira Porntaveetus, and Vorasuk Shotelersuk. Comparative transcriptome profiles of human dental pulp stem cells from maxillary and mandibular teeth. Scientific Reports, 12:8860, May 2022. doi:10.1038/s41598-022-12867-1.\">Faruangsaeng <em>et al.</em>, 2022</a>]</span> <span id=\"id5\">[<a class=\"reference internal\" href=\"../references.html#id6\" title=\"Sophie Pantalacci, Laurent Gu\u00e9guen, Coraline Petit, Anne Lambert, Renata Peterkov\u00e0, and Marie S\u00e9mon. Transcriptomic signatures shaped by cell proportions shed light on comparative developmental biology. Genome Biology, 18:29, February 2017. doi:10.1186/s13059-017-1157-7.\">Pantalacci <em>et al.</em>, 2017</a>]</span>. This is significant in mice, as the mandibular incisors of mice are longer than maxillary incisors.\n<img alt=\"png\" src=\"../_images/meta_tissue_tooth.png\"/>\nEven more significant is the distinction between <strong>tooth types</strong>. The differences between molars and incisors extend beyond shape to include enamel structure. In incisors, the enamel exhibits a uniserial lamellar pattern of prisms in the inner enamel, while the outer enamel features parallel radial prisms. In contrast, molars display a multifaceted rod pattern with more variable structures than those found in incisors.<span id=\"id6\">[<a class=\"reference internal\" href=\"../references.html#id7\" title=\"Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. eLife, 6:e24712, May 2017. doi:10.7554/eLife.24712.\">Seidel <em>et al.</em>, 2017</a>]</span></p>", "a[href=\"#data-accession\"]": "<h3 class=\"tippy-header\" style=\"margin-top: 0;\">Data accession<a class=\"headerlink\" href=\"#data-accession\" title=\"Permalink to this heading\">#</a></h3><p>This is the accession number of the data. For example, the data of <code class=\"docutils literal notranslate\"><span class=\"pre\">ToothNc_Junjun</span></code> is <a class=\"reference external\" href=\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE189381\">GSE189381</a>. Once you have recorded this information, you can easily access the data if you forget the data source.</p>", "a[href=\"#what-metadata-to-collect\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">What metadata to collect?<a class=\"headerlink\" href=\"#what-metadata-to-collect\" title=\"Permalink to this heading\">#</a></h2><p>While this is highly dependent on the study, here are some general categories of metadata that are often collected:</p>", "a[href=\"#id14\"]": "<figure class=\"align-center\" id=\"id14\">\n<a class=\"bg-primary reference internal image-reference\" href=\"../_images/20241023_enamel.png\"><img alt=\"fishy\" class=\"bg-primary\" src=\"../_images/20241023_enamel.png\" style=\"width: 800px;\"/></a>\n<figcaption>\n<p><span class=\"caption-text\">Comparison of incisor and molar enamel structure. Figure adapted from <span id=\"id7\">[<a class=\"reference internal\" href=\"../references.html#id7\" title=\"Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. eLife, 6:e24712, May 2017. doi:10.7554/eLife.24712.\">Seidel <em>et al.</em>, 2017</a>]</span>.</span><a class=\"headerlink\" href=\"#id14\" title=\"Permalink to this image\">#</a></p>\n</figcaption>\n</figure>", "a[href=\"#sequencing-platform\"]": "<h3 class=\"tippy-header\" style=\"margin-top: 0;\">Sequencing platform<a class=\"headerlink\" href=\"#sequencing-platform\" title=\"Permalink to this heading\">#</a></h3><p>Different platform may have huge impact on batch effect because of the different sequencing depth, handling personnel, reagent lots and equipments <span id=\"id12\">[<a class=\"reference internal\" href=\"../references.html#id12\" title=\"Hoa Thi Nhu Tran, Kok Siong Ang, Marion Chevrier, Xiaomeng Zhang, Nicole Yee Shin Lee, Michelle Goh, and Jinmiao Chen. A benchmark of batch-effect correction methods for single-cell RNA sequencing data. Genome Biology, 21(1):12, January 2020. doi:10.1186/s13059-019-1850-9.\">Tran <em>et al.</em>, 2020</a>]</span>. Even the same platform with different protocols, like 10Xv2 and 10Xv3, may also matters <span id=\"id13\">[<a class=\"reference internal\" href=\"../references.html#id13\" title=\"Lisa Sikkema, Ciro Ram\u00edrez-Su\u00e1stegui, Daniel C. Strobl, Tessa E. Gillett, Luke Zappia, Elo Madissoon, Nikolay S. Markov, Laure-Emmanuelle Zaragosi, Yuge Ji, Meshal Ansari, Marie-Jeanne Arguel, Leonie Apperloo, Martin Banchero, Christophe B\u00e9cavin, Marijn Berg, Evgeny Chichelnitskiy, Mei-i Chung, Antoine Collin, Aurore C. A. Gay, Janine Gote-Schniering, Baharak Hooshiar Kashani, Kemal Inecik, Manu Jain, Theodore S. Kapellos, Tessa M. Kole, Sylvie Leroy, Christoph H. Mayr, Amanda J. Oliver, Michael von Papen, Lance Peter, Chase J. Taylor, Thomas Walzthoeni, Chuan Xu, Linh T. Bui, Carlo De Donno, Leander Dony, Alen Faiz, Minzhe Guo, Austin J. Gutierrez, Lukas Heumos, Ni Huang, Ignacio L. Ibarra, Nathan D. Jackson, Preetish Kadur Lakshminarasimha Murthy, Mohammad Lotfollahi, Tracy Tabib, Carlos Talavera-L\u00f3pez, Kyle J. Travaglini, Anna Wilbrey-Clark, Kaylee B. Worlock, Masahiro Yoshida, Maarten van den Berge, Yohan Boss\u00e9, Tushar J. Desai, Oliver Eickelberg, Naftali Kaminski, Mark A. Krasnow, Robert Lafyatis, Marko Z. Nikolic, Joseph E. Powell, Jayaraj Rajagopal, Mauricio Rojas, Orit Rozenblatt-Rosen, Max A. Seibold, Dean Sheppard, Douglas P. Shepherd, Don D. Sin, Wim Timens, Alexander M. Tsankov, Jeffrey Whitsett, Yan Xu, Nicholas E. Banovich, Pascal Barbry, Thu Elizabeth Duong, Christine S. Falk, Kerstin B. Meyer, Jonathan A. Kropski, Dana Pe'er, Herbert B. Schiller, Purushothama Rao Tata, Joachim L. Schultze, Sara A. Teichmann, Alexander V. Misharin, Martijn C. Nawijn, Malte D. Luecken, and Fabian J. Theis. An integrated cell atlas of the lung in health and disease. Nature Medicine, 29(6):1563\u20131577, June 2023. doi:10.1038/s41591-023-02327-2.\">Sikkema <em>et al.</em>, 2023</a>]</span>. Therefore, we collect these information for each sample.</p>", "a[href=\"#introduction\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Introduction<a class=\"headerlink\" href=\"#introduction\" title=\"Permalink to this heading\">#</a></h2><p>When you\u2019re working on large-scale single-cell datasets, you should manually collect metadata for each study. Here are some benefits:</p>", "a[href=\"#study-metadata\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Study metadata<a class=\"headerlink\" href=\"#study-metadata\" title=\"Permalink to this heading\">#</a></h2><h3>Study name &amp; Sample name<a class=\"headerlink\" href=\"#study-name-sample-name\" title=\"Permalink to this heading\">#</a></h3><p>Usually, we define the study as one publication conducted by one lab. Samples in the same study are usually comparable. For example, the paper <a class=\"reference external\" href=\"https://www.nature.com/articles/s41467-022-32490-y\"><strong>Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis</strong></a> <span id=\"id1\">[<a class=\"reference internal\" href=\"../references.html#id2\" title=\"Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. Nature Communications, 13(1):4803, August 2022. doi:10.1038/s41467-022-32490-y.\">Jing <em>et al.</em>, 2022</a>]</span> conducted by Yang Chai lab contains five samples ranging from E13.5 to PN7.5 in mouse embryo development. In my convention, I use characters \u201cuniqueID+author\u201d to name the study and use \u201cuniqueID+sampleName\u201d to name the sample. In this example, the study name is <code class=\"docutils literal notranslate\"><span class=\"pre\">ToothNc_Junjun</span></code> and the sample name is <code class=\"docutils literal notranslate\"><span class=\"pre\">ToothNc_Junjun_E13.5</span></code>. The most important thing is the study name should be easy to remember and unique.</p>", "a[href=\"#technical-covariates\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Technical covariates<a class=\"headerlink\" href=\"#technical-covariates\" title=\"Permalink to this heading\">#</a></h2><h3>Sequencing platform<a class=\"headerlink\" href=\"#sequencing-platform\" title=\"Permalink to this heading\">#</a></h3><p>Different platform may have huge impact on batch effect because of the different sequencing depth, handling personnel, reagent lots and equipments <span id=\"id12\">[<a class=\"reference internal\" href=\"../references.html#id12\" title=\"Hoa Thi Nhu Tran, Kok Siong Ang, Marion Chevrier, Xiaomeng Zhang, Nicole Yee Shin Lee, Michelle Goh, and Jinmiao Chen. A benchmark of batch-effect correction methods for single-cell RNA sequencing data. Genome Biology, 21(1):12, January 2020. doi:10.1186/s13059-019-1850-9.\">Tran <em>et al.</em>, 2020</a>]</span>. Even the same platform with different protocols, like 10Xv2 and 10Xv3, may also matters <span id=\"id13\">[<a class=\"reference internal\" href=\"../references.html#id13\" title=\"Lisa Sikkema, Ciro Ram\u00edrez-Su\u00e1stegui, Daniel C. Strobl, Tessa E. Gillett, Luke Zappia, Elo Madissoon, Nikolay S. Markov, Laure-Emmanuelle Zaragosi, Yuge Ji, Meshal Ansari, Marie-Jeanne Arguel, Leonie Apperloo, Martin Banchero, Christophe B\u00e9cavin, Marijn Berg, Evgeny Chichelnitskiy, Mei-i Chung, Antoine Collin, Aurore C. A. Gay, Janine Gote-Schniering, Baharak Hooshiar Kashani, Kemal Inecik, Manu Jain, Theodore S. Kapellos, Tessa M. Kole, Sylvie Leroy, Christoph H. Mayr, Amanda J. Oliver, Michael von Papen, Lance Peter, Chase J. Taylor, Thomas Walzthoeni, Chuan Xu, Linh T. Bui, Carlo De Donno, Leander Dony, Alen Faiz, Minzhe Guo, Austin J. Gutierrez, Lukas Heumos, Ni Huang, Ignacio L. Ibarra, Nathan D. Jackson, Preetish Kadur Lakshminarasimha Murthy, Mohammad Lotfollahi, Tracy Tabib, Carlos Talavera-L\u00f3pez, Kyle J. Travaglini, Anna Wilbrey-Clark, Kaylee B. Worlock, Masahiro Yoshida, Maarten van den Berge, Yohan Boss\u00e9, Tushar J. Desai, Oliver Eickelberg, Naftali Kaminski, Mark A. Krasnow, Robert Lafyatis, Marko Z. Nikolic, Joseph E. Powell, Jayaraj Rajagopal, Mauricio Rojas, Orit Rozenblatt-Rosen, Max A. Seibold, Dean Sheppard, Douglas P. Shepherd, Don D. Sin, Wim Timens, Alexander M. Tsankov, Jeffrey Whitsett, Yan Xu, Nicholas E. Banovich, Pascal Barbry, Thu Elizabeth Duong, Christine S. Falk, Kerstin B. Meyer, Jonathan A. Kropski, Dana Pe'er, Herbert B. Schiller, Purushothama Rao Tata, Joachim L. Schultze, Sara A. Teichmann, Alexander V. Misharin, Martijn C. Nawijn, Malte D. Luecken, and Fabian J. Theis. An integrated cell atlas of the lung in health and disease. Nature Medicine, 29(6):1563\u20131577, June 2023. doi:10.1038/s41591-023-02327-2.\">Sikkema <em>et al.</em>, 2023</a>]</span>. Therefore, we collect these information for each sample.</p>", "a[href=\"#biological-covariates\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Biological covariates<a class=\"headerlink\" href=\"#biological-covariates\" title=\"Permalink to this heading\">#</a></h2><h3>Tissue information<a class=\"headerlink\" href=\"#tissue-information\" title=\"Permalink to this heading\">#</a></h3><p>This is the most tricky part, and it is highly dependent on your purpose. For example, when I build the <a class=\"reference external\" href=\"https://trajatlas.readthedocs.io/en/stable/introduction/Differentiation_Atlas.html\">Differentiation Atlas</a> focused on bone development, I found it\u2019s hard to use one single tissue information to summarize the tissue information. So I use multi-level tissue information for description. Some reviews <span id=\"id2\">[<a class=\"reference internal\" href=\"../references.html#id3\" title=\"Toshihide Mizoguchi and Noriaki Ono. The diverse origin of bone-forming osteoblasts. Journal of Bone and Mineral Research, 36(8):1432\u20131447, 2021. doi:10.1002/jbmr.4410.\">Mizoguchi and Ono, 2021</a>]</span> suggest there are three main organs that generate the osteoblasts: Head, Limb bud, and Long bone, and the osteogenic pathways are significantly different across the organs. So I used <strong>tissue origin</strong> which contains these three categories to describe the main organ sources. And in a finer resolution, we can see that the organ can be divided into the <strong>histological</strong> and <strong>tissue location</strong> subtypes <span id=\"id3\">[<a class=\"reference internal\" href=\"../references.html#id4\" title=\"Hui Sophie Shu, Yiming Liam Liu, Xinyu Thomas Tang, Xinyi Shirley Zhang, Bin Zhou, Weiguo Zou, and Bo O. Zhou. Tracing the skeletal progenitor transition during postnatal bone formation. Cell Stem Cell, 28(12):2122-2136.e3, December 2021. doi:10.1016/j.stem.2021.08.010.\">Shu <em>et al.</em>, 2021</a>]</span>. For example, the long bone can be divided into metaphysis, diaphysis, and epiphysis, which are histologically different. And it can also be divided into tibia, femur, which is location different. Therefore, I finally form three-level tissue metadata.\n<img alt=\"png\" src=\"../_images/meta_tissue.png\"/></p><p>And in tooth development, the situation is completely different. First, the tooth from <strong>mandibular or maxillary</strong> may have great shape and transcriptional differences <span id=\"id4\">[<a class=\"reference internal\" href=\"../references.html#id5\" title=\"Thira Faruangsaeng, Sermporn Thaweesapphitak, Chompak Khamwachirapitak, Thantrira Porntaveetus, and Vorasuk Shotelersuk. Comparative transcriptome profiles of human dental pulp stem cells from maxillary and mandibular teeth. Scientific Reports, 12:8860, May 2022. doi:10.1038/s41598-022-12867-1.\">Faruangsaeng <em>et al.</em>, 2022</a>]</span> <span id=\"id5\">[<a class=\"reference internal\" href=\"../references.html#id6\" title=\"Sophie Pantalacci, Laurent Gu\u00e9guen, Coraline Petit, Anne Lambert, Renata Peterkov\u00e0, and Marie S\u00e9mon. Transcriptomic signatures shaped by cell proportions shed light on comparative developmental biology. Genome Biology, 18:29, February 2017. doi:10.1186/s13059-017-1157-7.\">Pantalacci <em>et al.</em>, 2017</a>]</span>. This is significant in mice, as the mandibular incisors of mice are longer than maxillary incisors.\n<img alt=\"png\" src=\"../_images/meta_tissue_tooth.png\"/>\nEven more significant is the distinction between <strong>tooth types</strong>. The differences between molars and incisors extend beyond shape to include enamel structure. In incisors, the enamel exhibits a uniserial lamellar pattern of prisms in the inner enamel, while the outer enamel features parallel radial prisms. In contrast, molars display a multifaceted rod pattern with more variable structures than those found in incisors.<span id=\"id6\">[<a class=\"reference internal\" href=\"../references.html#id7\" title=\"Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. eLife, 6:e24712, May 2017. doi:10.7554/eLife.24712.\">Seidel <em>et al.</em>, 2017</a>]</span></p>", "a[href=\"#age\"]": "<h3 class=\"tippy-header\" style=\"margin-top: 0;\">Age<a class=\"headerlink\" href=\"#age\" title=\"Permalink to this heading\">#</a></h3><p>Another important biological covariate is <strong>age</strong>. Based on the shape of tooth germ, we can roughly divide the tooth development into five stages: thicking, bud, cap, bell and erupted. <span id=\"id10\">[<a class=\"reference internal\" href=\"../references.html#id10\" title=\"Abigail Tucker and Paul Sharpe. The cutting-edge of mammalian development; how the embryo makes teeth. Nature Reviews Genetics, 5(7):499\u2013508, July 2004. doi:10.1038/nrg1380.\">Tucker and Sharpe, 2004</a>]</span>\n<img alt=\"png\" src=\"../_images/meta_age.png\"/>\nStudies also report the age-related difference in the dental pulp. <span id=\"id11\">[<a class=\"reference internal\" href=\"../references.html#id11\" title=\"Wei Wu, Jian Zhou, Chong-Tao Xu, Jie Zhang, Yan-Jiao Jin, and Geng-Lin Sun. Derivation and growth characteristics of dental pulp stem cells from patients of different ages. Molecular Medicine Reports, 12(4):5127\u20135134, October 2015. doi:10.3892/mmr.2015.4106.\">Wu <em>et al.</em>, 2015</a>]</span> Therefore, we also collect the detailed age information for each sample and base on the age we divided into different stages.\n<img alt=\"png\" src=\"../_images/meta_age_stage.png\"/></p>"}
skip_classes = ["headerlink", "sd-stretched-link"]

window.onload = function () {
    for (const [select, tip_html] of Object.entries(selector_to_html)) {
        const links = document.querySelectorAll(`div.content ${select}`);
        for (const link of links) {
            if (skip_classes.some(c => link.classList.contains(c))) {
                continue;
            }

            tippy(link, {
                content: tip_html,
                allowHTML: true,
                arrow: true,
                placement: 'auto-start', maxWidth: 500, interactive: false,
                onShow(instance) {MathJax.typesetPromise([instance.popper]).then(() => {});},
            });
        };
    };
    console.log("tippy tips loaded!");
};
