<!doctype html>
<html class="no-js" lang="en" data-content_root="">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="First round preprocess" href="first_round_preprocess.html" /><link rel="prev" title="Introduction to scAtlas" href="../Introduction/introduction.html" />

    <!-- Generated with Sphinx 7.1.2 and Furo 2024.01.29 -->
        <title>Metadata Collection - scAtlas 0.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?v=135e06be" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css?v=0a3b3ea7" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?v=36a5483c" />
    <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css" />
    <link rel="stylesheet" type="text/css" href="../_static/css/override.css" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #003262;
  --color-brand-content: #003262;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #272822;
  --color-code-foreground: #f8f8f2;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #272822;
  --color-code-foreground: #f8f8f2;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">scAtlas 0.0.1 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo only-light" src="../_static/tooth_log.png" alt="Light Logo"/>
    <img class="sidebar-logo only-dark" src="../_static/tooth_log.png" alt="Dark Logo"/>
  </div>
  
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Introduction/introduction.html">Introduction to scAtlas</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Metadata Collection</a></li>
<li class="toctree-l1"><a class="reference internal" href="first_round_preprocess.html">First round preprocess</a></li>
<li class="toctree-l1"><a class="reference internal" href="droplet.html">Droplet discovery</a></li>
<li class="toctree-l1"><a class="reference internal" href="first_round_annotation.html">First round annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="post_annotation.html">Post annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="batch_division.html">Batch division</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pre-integration/hypertune.html">Hyperparameter tuning</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pre-integration/scib-benchmark.html">Integration methods benchmarking</a></li>
<li class="toctree-l1"><a class="reference internal" href="../integration/20241024_coarse_label.html">Cluster the datasets to get first level annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../about/team.html">Team</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about/cite.html">Citing scAtlas</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributing.html">Contributing Guide</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="metadata-collection">
<h1>Metadata Collection<a class="headerlink" href="#metadata-collection" title="Permalink to this heading">#</a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this heading">#</a></h2>
<p>When you’re working on large-scale single-cell datasets, you should manually collect metadata for each study. Here are some benefits:</p>
<ul class="simple">
<li><p>It allows you to build a standard criterion to compare across studies.</p></li>
<li><p>It allows you to conveniently add study-level annotations to cell-level metadata.</p></li>
<li><p>It allows you to access the information you need quickly.</p></li>
</ul>
</section>
<section id="what-metadata-to-collect">
<h2>What metadata to collect?<a class="headerlink" href="#what-metadata-to-collect" title="Permalink to this heading">#</a></h2>
<p>While this is highly dependent on the study, here are some general categories of metadata that are often collected:</p>
<ul class="simple">
<li><p>Study metadata:</p>
<ul>
<li><p>Study name</p></li>
<li><p>Sample name</p></li>
<li><p>Data accession</p></li>
<li><p>Study publication journal</p></li>
<li><p>Study publication year</p></li>
</ul>
</li>
<li><p>Biological covariates:</p>
<ul>
<li><p>Tissue information</p></li>
<li><p>Age</p></li>
<li><p>Gene type</p></li>
<li><p>Treatment</p></li>
<li><p>Disease</p></li>
<li><p>Organism</p></li>
</ul>
</li>
<li><p>Technical covariates:</p>
<ul>
<li><p>Sequencing platform</p></li>
<li><p>Dissociation enzyme</p></li>
</ul>
</li>
</ul>
<p>In the following sections, we will go through an example of the metadata we collected in the tooth atlas.</p>
</section>
<section id="study-metadata">
<h2>Study metadata<a class="headerlink" href="#study-metadata" title="Permalink to this heading">#</a></h2>
<section id="study-name-sample-name">
<h3>Study name &amp; Sample name<a class="headerlink" href="#study-name-sample-name" title="Permalink to this heading">#</a></h3>
<p>Usually, we define the study as one publication conducted by one lab. Samples in the same study are usually comparable. For example, the paper <a class="reference external" href="https://www.nature.com/articles/s41467-022-32490-y"><strong>Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis</strong></a> <span id="id1">[<a class="reference internal" href="../references.html#id2" title="Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. Nature Communications, 13(1):4803, August 2022. doi:10.1038/s41467-022-32490-y.">Jing <em>et al.</em>, 2022</a>]</span> conducted by Yang Chai lab contains five samples ranging from E13.5 to PN7.5 in mouse embryo development. In my convention, I use characters “uniqueID+author” to name the study and use “uniqueID+sampleName” to name the sample. In this example, the study name is <code class="docutils literal notranslate"><span class="pre">ToothNc_Junjun</span></code> and the sample name is <code class="docutils literal notranslate"><span class="pre">ToothNc_Junjun_E13.5</span></code>. The most important thing is the study name should be easy to remember and unique.</p>
</section>
<section id="data-accession">
<h3>Data accession<a class="headerlink" href="#data-accession" title="Permalink to this heading">#</a></h3>
<p>This is the accession number of the data. For example, the data of <code class="docutils literal notranslate"><span class="pre">ToothNc_Junjun</span></code> is <a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE189381">GSE189381</a>. Once you have recorded this information, you can easily access the data if you forget the data source.</p>
</section>
</section>
<section id="biological-covariates">
<h2>Biological covariates<a class="headerlink" href="#biological-covariates" title="Permalink to this heading">#</a></h2>
<section id="tissue-information">
<h3>Tissue information<a class="headerlink" href="#tissue-information" title="Permalink to this heading">#</a></h3>
<p>This is the most tricky part, and it is highly dependent on your purpose. For example, when I build the <a class="reference external" href="https://trajatlas.readthedocs.io/en/stable/introduction/Differentiation_Atlas.html">Differentiation Atlas</a> focused on bone development, I found it’s hard to use one single tissue information to summarize the tissue information. So I use multi-level tissue information for description. Some reviews <span id="id2">[<a class="reference internal" href="../references.html#id3" title="Toshihide Mizoguchi and Noriaki Ono. The diverse origin of bone-forming osteoblasts. Journal of Bone and Mineral Research, 36(8):1432–1447, 2021. doi:10.1002/jbmr.4410.">Mizoguchi and Ono, 2021</a>]</span> suggest there are three main organs that generate the osteoblasts: Head, Limb bud, and Long bone, and the osteogenic pathways are significantly different across the organs. So I used <strong>tissue origin</strong> which contains these three categories to describe the main organ sources. And in a finer resolution, we can see that the organ can be divided into the <strong>histological</strong> and <strong>tissue location</strong> subtypes <span id="id3">[<a class="reference internal" href="../references.html#id4" title="Hui Sophie Shu, Yiming Liam Liu, Xinyu Thomas Tang, Xinyi Shirley Zhang, Bin Zhou, Weiguo Zou, and Bo O. Zhou. Tracing the skeletal progenitor transition during postnatal bone formation. Cell Stem Cell, 28(12):2122-2136.e3, December 2021. doi:10.1016/j.stem.2021.08.010.">Shu <em>et al.</em>, 2021</a>]</span>. For example, the long bone can be divided into metaphysis, diaphysis, and epiphysis, which are histologically different. And it can also be divided into tibia, femur, which is location different. Therefore, I finally form three-level tissue metadata.
<img alt="png" src="../_images/meta_tissue.png" /></p>
<p>And in tooth development, the situation is completely different. First, the tooth from <strong>mandibular or maxillary</strong> may have great shape and transcriptional differences <span id="id4">[<a class="reference internal" href="../references.html#id5" title="Thira Faruangsaeng, Sermporn Thaweesapphitak, Chompak Khamwachirapitak, Thantrira Porntaveetus, and Vorasuk Shotelersuk. Comparative transcriptome profiles of human dental pulp stem cells from maxillary and mandibular teeth. Scientific Reports, 12:8860, May 2022. doi:10.1038/s41598-022-12867-1.">Faruangsaeng <em>et al.</em>, 2022</a>]</span> <span id="id5">[<a class="reference internal" href="../references.html#id6" title="Sophie Pantalacci, Laurent Guéguen, Coraline Petit, Anne Lambert, Renata Peterkovà, and Marie Sémon. Transcriptomic signatures shaped by cell proportions shed light on comparative developmental biology. Genome Biology, 18:29, February 2017. doi:10.1186/s13059-017-1157-7.">Pantalacci <em>et al.</em>, 2017</a>]</span>. This is significant in mice, as the mandibular incisors of mice are longer than maxillary incisors.
<img alt="png" src="../_images/meta_tissue_tooth.png" />
Even more significant is the distinction between <strong>tooth types</strong>. The differences between molars and incisors extend beyond shape to include enamel structure. In incisors, the enamel exhibits a uniserial lamellar pattern of prisms in the inner enamel, while the outer enamel features parallel radial prisms. In contrast, molars display a multifaceted rod pattern with more variable structures than those found in incisors.<span id="id6">[<a class="reference internal" href="../references.html#id7" title="Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. eLife, 6:e24712, May 2017. doi:10.7554/eLife.24712.">Seidel <em>et al.</em>, 2017</a>]</span></p>
<figure class="align-center" id="id14">
<a class="bg-primary reference internal image-reference" href="../_images/20241023_enamel.png"><img alt="fishy" class="bg-primary" src="../_images/20241023_enamel.png" style="width: 800px;" /></a>
<figcaption>
<p><span class="caption-text">Comparison of incisor and molar enamel structure. Figure adapted from <span id="id7">[<a class="reference internal" href="../references.html#id7" title="Kerstin Seidel, Pauline Marangoni, Cynthia Tang, Bahar Houshmand, Wen Du, Richard L Maas, Steven Murray, Michael C Oldham, and Ophir D Klein. Resolving stem and progenitor cells in the adult mouse incisor through gene co-expression analysis. eLife, 6:e24712, May 2017. doi:10.7554/eLife.24712.">Seidel <em>et al.</em>, 2017</a>]</span>.</span><a class="headerlink" href="#id14" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<p>Some studies may focus on a specific <strong>tooth histology type</strong>. For example the study <span id="id8">[<a class="reference internal" href="../references.html#id8" title="Jan Krivanek, Ruslan A. Soldatov, Maria Eleni Kastriti, Tatiana Chontorotzea, Anna Nele Herdina, Julian Petersen, Bara Szarowska, Marie Landova, Veronika Kovar Matejova, Lydie Izakovicova Holla, Ulrike Kuchler, Ivana Vidovic Zdrilic, Anushree Vijaykumar, Anamaria Balic, Pauline Marangoni, Ophir D. Klein, Vitor C. M. Neves, Val Yianni, Paul T. Sharpe, Tibor Harkany, Brian D. Metscher, Marc Bajénoff, Mina Mina, Kaj Fried, Peter V. Kharchenko, and Igor Adameyko. Dental cell type atlas reveals stem and differentiated cell types in mouse and human teeth. Nature Communications, 11(1):4816, September 2020. doi:10.1038/s41467-020-18512-7.">Krivanek <em>et al.</em>, 2020</a>]</span> use the dental pulp as the tissue source, while the study <span id="id9">[<a class="reference internal" href="../references.html#id9" title="S.Y. Luo, S. Wang, Z.X. Liu, Q. Bian, and X.D. Wang. Six1 Regulates Mouse Incisor Development by Promoting Dlx1/2/5 Expression. Journal of Dental Research, 103(10):1017–1027, September 2024. doi:10.1177/00220345241256286.">Luo <em>et al.</em>, 2024</a>]</span> focus on the tooth germ.</p>
</section>
<section id="age">
<h3>Age<a class="headerlink" href="#age" title="Permalink to this heading">#</a></h3>
<p>Another important biological covariate is <strong>age</strong>. Based on the shape of tooth germ, we can roughly divide the tooth development into five stages: thicking, bud, cap, bell and erupted. <span id="id10">[<a class="reference internal" href="../references.html#id10" title="Abigail Tucker and Paul Sharpe. The cutting-edge of mammalian development; how the embryo makes teeth. Nature Reviews Genetics, 5(7):499–508, July 2004. doi:10.1038/nrg1380.">Tucker and Sharpe, 2004</a>]</span>
<img alt="png" src="../_images/meta_age.png" />
Studies also report the age-related difference in the dental pulp. <span id="id11">[<a class="reference internal" href="../references.html#id11" title="Wei Wu, Jian Zhou, Chong-Tao Xu, Jie Zhang, Yan-Jiao Jin, and Geng-Lin Sun. Derivation and growth characteristics of dental pulp stem cells from patients of different ages. Molecular Medicine Reports, 12(4):5127–5134, October 2015. doi:10.3892/mmr.2015.4106.">Wu <em>et al.</em>, 2015</a>]</span> Therefore, we also collect the detailed age information for each sample and base on the age we divided into different stages.
<img alt="png" src="../_images/meta_age_stage.png" /></p>
</section>
</section>
<section id="technical-covariates">
<h2>Technical covariates<a class="headerlink" href="#technical-covariates" title="Permalink to this heading">#</a></h2>
<section id="sequencing-platform">
<h3>Sequencing platform<a class="headerlink" href="#sequencing-platform" title="Permalink to this heading">#</a></h3>
<p>Different platform may have huge impact on batch effect because of the different sequencing depth, handling personnel, reagent lots and equipments <span id="id12">[<a class="reference internal" href="../references.html#id12" title="Hoa Thi Nhu Tran, Kok Siong Ang, Marion Chevrier, Xiaomeng Zhang, Nicole Yee Shin Lee, Michelle Goh, and Jinmiao Chen. A benchmark of batch-effect correction methods for single-cell RNA sequencing data. Genome Biology, 21(1):12, January 2020. doi:10.1186/s13059-019-1850-9.">Tran <em>et al.</em>, 2020</a>]</span>. Even the same platform with different protocols, like 10Xv2 and 10Xv3, may also matters <span id="id13">[<a class="reference internal" href="../references.html#id13" title="Lisa Sikkema, Ciro Ramírez-Suástegui, Daniel C. Strobl, Tessa E. Gillett, Luke Zappia, Elo Madissoon, Nikolay S. Markov, Laure-Emmanuelle Zaragosi, Yuge Ji, Meshal Ansari, Marie-Jeanne Arguel, Leonie Apperloo, Martin Banchero, Christophe Bécavin, Marijn Berg, Evgeny Chichelnitskiy, Mei-i Chung, Antoine Collin, Aurore C. A. Gay, Janine Gote-Schniering, Baharak Hooshiar Kashani, Kemal Inecik, Manu Jain, Theodore S. Kapellos, Tessa M. Kole, Sylvie Leroy, Christoph H. Mayr, Amanda J. Oliver, Michael von Papen, Lance Peter, Chase J. Taylor, Thomas Walzthoeni, Chuan Xu, Linh T. Bui, Carlo De Donno, Leander Dony, Alen Faiz, Minzhe Guo, Austin J. Gutierrez, Lukas Heumos, Ni Huang, Ignacio L. Ibarra, Nathan D. Jackson, Preetish Kadur Lakshminarasimha Murthy, Mohammad Lotfollahi, Tracy Tabib, Carlos Talavera-López, Kyle J. Travaglini, Anna Wilbrey-Clark, Kaylee B. Worlock, Masahiro Yoshida, Maarten van den Berge, Yohan Bossé, Tushar J. Desai, Oliver Eickelberg, Naftali Kaminski, Mark A. Krasnow, Robert Lafyatis, Marko Z. Nikolic, Joseph E. Powell, Jayaraj Rajagopal, Mauricio Rojas, Orit Rozenblatt-Rosen, Max A. Seibold, Dean Sheppard, Douglas P. Shepherd, Don D. Sin, Wim Timens, Alexander M. Tsankov, Jeffrey Whitsett, Yan Xu, Nicholas E. Banovich, Pascal Barbry, Thu Elizabeth Duong, Christine S. Falk, Kerstin B. Meyer, Jonathan A. Kropski, Dana Pe'er, Herbert B. Schiller, Purushothama Rao Tata, Joachim L. Schultze, Sara A. Teichmann, Alexander V. Misharin, Martijn C. Nawijn, Malte D. Luecken, and Fabian J. Theis. An integrated cell atlas of the lung in health and disease. Nature Medicine, 29(6):1563–1577, June 2023. doi:10.1038/s41591-023-02327-2.">Sikkema <em>et al.</em>, 2023</a>]</span>. Therefore, we collect these information for each sample.</p>
</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="first_round_preprocess.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">First round preprocess</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../Introduction/introduction.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Introduction to scAtlas</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Gilbert Han
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link fab fa-github" href="https://github.com/GilbertHan1011/toothAtlasManuscript" aria-label="GitHub"></a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Metadata Collection</a><ul>
<li><a class="reference internal" href="#introduction">Introduction</a></li>
<li><a class="reference internal" href="#what-metadata-to-collect">What metadata to collect?</a></li>
<li><a class="reference internal" href="#study-metadata">Study metadata</a><ul>
<li><a class="reference internal" href="#study-name-sample-name">Study name &amp; Sample name</a></li>
<li><a class="reference internal" href="#data-accession">Data accession</a></li>
</ul>
</li>
<li><a class="reference internal" href="#biological-covariates">Biological covariates</a><ul>
<li><a class="reference internal" href="#tissue-information">Tissue information</a></li>
<li><a class="reference internal" href="#age">Age</a></li>
</ul>
</li>
<li><a class="reference internal" href="#technical-covariates">Technical covariates</a><ul>
<li><a class="reference internal" href="#sequencing-platform">Sequencing platform</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js?v=f6245a2f"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="../_static/scripts/furo.js?v=32e29ea5"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/design-tabs.js?v=36754332"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script defer="defer" src="https://unpkg.com/@popperjs/core@2"></script>
    <script defer="defer" src="https://unpkg.com/tippy.js@6"></script>
    <script defer="defer" src="../_static/tippy/preprocess/metadata_process.9634c7d2-e300-4ebe-8745-688b324dee69.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    </body>
</html>