selector_to_html = {"a[href=\"Level2_anno.html#apical-and-distal-pulp\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Apical and Distal pulp<a class=\"headerlink\" href=\"#apical-and-distal-pulp\" title=\"Permalink to this heading\">#</a></h2><p>Postnatal mesenchyme in <a class=\"reference internal\" href=\"Level1_anno.html\"><span class=\"std std-doc\">level 1</span></a> annotation can be devided into two main clusters: C9-6 and C9-7. The most important factors that drive the heterogeneity of postnatal pulp may be the spatial location. Apical pulp and distal pulp are have different gene expression profiles <span id=\"id5\">[<a class=\"reference internal\" href=\"../references.html#id29\" title=\"Florian Hermans, Celine Bueds, Lara Hemeryck, Ivo Lambrichts, Annelies Bronckaers, and Hugo Vankelecom. Establishment of inclusive single-cell transcriptome atlases from mouse and human tooth as powerful resource for dental research. Frontiers in Cell and Developmental Biology, 2022.\">Hermans <em>et al.</em>, 2022</a>]</span> <span id=\"id6\">[<a class=\"reference internal\" href=\"../references.html#id30\" title=\"Yingchun Zheng, Ting Lu, Leitao Zhang, Zhongzhi Gan, Aoxi Li, Chuandong He, Fei He, Sha He, Jian Zhang, and Fu Xiong. Single-cell RNA-seq analysis of rat molars reveals cell identity and driver genes associated with dental mesenchymal cell differentiation. BMC Biology, 22(1):1\u201313, September 2024. doi:10.1186/s12915-024-01996-w.\">Zheng <em>et al.</em>, 2024</a>]</span>.\nWe checked the expression of the marker genes of apical and distal pulp in these two clusters, which align with our expectation.</p><p><strong>Apical Pulp</strong></p>", "a[href=\"#annotation\"]": "<h1 class=\"tippy-header\" style=\"margin-top: 0;\">Annotation<a class=\"headerlink\" href=\"#annotation\" title=\"Permalink to this heading\">#</a></h1><p>In this section, we will adapt the annotation pipeline from <a class=\"reference external\" href=\"https://github.com/lsteuernagel/scHarmonization/\">scHarmonization</a> <span id=\"id1\">[<a class=\"reference internal\" href=\"../references.html#id22\" title=\"Lukas Steuernagel, Brian Y. H. Lam, Paul Klemm, Georgina K. C. Dowsett, Corinna A. Bauder, John A. Tadross, Tamara Sotelo Hitschfeld, Almudena del Rio Martin, Weiyi Chen, Alain J. de Solis, Henning Fenselau, Peter Davidsen, Irene Cimino, Sara N. Kohnke, Debra Rimmington, Anthony P. Coll, Andreas Beyer, Giles S. H. Yeo, and Jens C. Br\u00fcning. HypoMap\u2014a unified single-cell gene expression atlas of the murine hypothalamus. Nature Metabolism, 4(10):1402\u20131419, October 2022. doi:10.1038/s42255-022-00657-y.\">Steuernagel <em>et al.</em>, 2022</a>]</span> to our data.\nYou can just run this pipeline directly. Here, I just break down the pipeline into several steps for better understanding.</p>", "a[href=\"Level2_anno.html#motivation\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Motivation<a class=\"headerlink\" href=\"#motivation\" title=\"Permalink to this heading\">#</a></h2><p>In level 2 annotation, we devided the mesenchyme into 9 clusters, including four large populations (\u201cC9-1\u201d,\u201dC9-2\u201d,\u201dC9-6\u201d,\u201dC9-7\u201d) and five small populations. In this level, we want to assign the consensus annotation from literature to these clusters.\n<img alt=\"png\" src=\"../_images/anno_C9.png\"/></p>", "a[href=\"Level2_anno.html#odontoblast\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Odontoblast<a class=\"headerlink\" href=\"#odontoblast\" title=\"Permalink to this heading\">#</a></h2><p>Base on the expression of Phex, Smpd3, we can easily identify the odontoblast cluster C9-8.\n<img alt=\"png\" src=\"../_images/C9-8.png\"/></p>", "a[href=\"Level2_anno.html\"]": "<h1 class=\"tippy-header\" style=\"margin-top: 0;\">Level 2 annotation<a class=\"headerlink\" href=\"#level-2-annotation\" title=\"Permalink to this heading\">#</a></h1><h2>Motivation<a class=\"headerlink\" href=\"#motivation\" title=\"Permalink to this heading\">#</a></h2><p>In level 2 annotation, we devided the mesenchyme into 9 clusters, including four large populations (\u201cC9-1\u201d,\u201dC9-2\u201d,\u201dC9-6\u201d,\u201dC9-7\u201d) and five small populations. In this level, we want to assign the consensus annotation from literature to these clusters.\n<img alt=\"png\" src=\"../_images/anno_C9.png\"/></p>", "a[href=\"Level2_anno.html#bud-mesenchyme-cap-mesenchyme\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Bud Mesenchyme &amp; Cap Mesenchyme<a class=\"headerlink\" href=\"#bud-mesenchyme-cap-mesenchyme\" title=\"Permalink to this heading\">#</a></h2><p>In the study focused on tooth germ development <span id=\"id1\">[<a class=\"reference internal\" href=\"../references.html#id27\" title=\"Yaofeng Wang, Yifan Zhao, Shubin Chen, Xiaoming Chen, Yanmei Zhang, Hong Chen, Yuansong Liao, Jiashu Zhang, Di Wu, Hongxing Chu, Hongying Huang, Caixia Wu, Shijuan Huang, Huichao Xu, Bei Jia, Jie Liu, Bo Feng, Zhonghan Li, Dajiang Qin, Duanqing Pei, and Jinglei Cai. Single cell atlas of developing mouse dental germs reveals populations of CD24+ and Plac8+ odontogenic cells. Science Bulletin, 67(11):1154\u20131169, June 2022. doi:10.1016/j.scib.2022.03.012.\">Wang <em>et al.</em>, 2022</a>]</span> <span id=\"id2\">[<a class=\"reference internal\" href=\"../references.html#id2\" title=\"Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. Nature Communications, 13(1):4803, August 2022. doi:10.1038/s41467-022-32490-y.\">Jing <em>et al.</em>, 2022</a>]</span>, the data reveal significant heterogeneity among the datasets at different stages, such as the bud stage and cap stage. We have also observed that the thickening stage mesenchyme (&lt; E13.5) and bud stage (E13.5-E14.5) (See <a class=\"reference internal\" href=\"../preprocess/metadata_process.html\"><span class=\"std std-doc\">Metadata Collection</span></a>) are predominantly present in the C9-1 cluster. The C9-2 cluster is primarily composed of mesenchyme at the cap stage (E14.5-E18.5). Therefore, we assigned the C9-1 cluster as \u201cBud Mesenchyme\u201d and the C9-2 cluster as \u201cCap Mesenchyme\u201d.\n<img alt=\"png\" src=\"../_images/C9_age_bar.png\"/></p><p>Bud mesenchyme highly expressed the genes related to embroyic develpment, such as <a class=\"reference external\" href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=tbx18\">Tbx18</a> and <a class=\"reference external\" href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=hic1\">Hic1</a> <span id=\"id3\">[<a class=\"reference internal\" href=\"../references.html#id28\" title=\"Martin Arostegui, R. Wilder Scott, Kerstin B\u00f6se, and T. Michael Underhill. Cellular taxonomy of Hic1+ mesenchymal progenitor derivatives in the limb: from embryo to adult. Nature Communications, 13(1):4989, August 2022. doi:10.1038/s41467-022-32695-1.\">Arostegui <em>et al.</em>, 2022</a>]</span>.\n<img alt=\"png\" src=\"../_images/C9-1-gene.png\"/></p>"}
skip_classes = ["headerlink", "sd-stretched-link"]

window.onload = function () {
    for (const [select, tip_html] of Object.entries(selector_to_html)) {
        const links = document.querySelectorAll(`div.content ${select}`);
        for (const link of links) {
            if (skip_classes.some(c => link.classList.contains(c))) {
                continue;
            }

            tippy(link, {
                content: tip_html,
                allowHTML: true,
                arrow: true,
                placement: 'auto-start', maxWidth: 500, interactive: false,
                onShow(instance) {MathJax.typesetPromise([instance.popper]).then(() => {});},
            });
        };
    };
    console.log("tippy tips loaded!");
};
