selector_to_html = {"a[href=\"#low-quality\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Low quality<a class=\"headerlink\" href=\"#low-quality\" title=\"Permalink to this heading\">#</a></h2><p>We have observed a cluster (C9-5) that has low quality, which is likely due to the presence of a large number of RBCs in the sample.\n<img alt=\"png\" src=\"../_images/C9-5.png\"/></p>", "a[href=\"#epithelium-like\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Epithelium-like<a class=\"headerlink\" href=\"#epithelium-like\" title=\"Permalink to this heading\">#</a></h2><p>We observed a cluster (C9-9) that highly expresses genes related to various cell types, such as Krt14 (epithelium-related), Ambn (ameloblast-related), Prrx1 (mesenchyme-related), Igfbp5 (dental follicle-related), and Bglap (osteoblast-related) <span id=\"id16\">[<a class=\"reference internal\" href=\"../references.html#id19\" title=\"Litian Han, Yaoting Ji, Yiqian Yu, Yueqi Ni, Hao Zeng, Xiaoxin Zhang, Huan Liu, and Yufeng Zhang. Trajectory-centric framework TrajAtlas reveals multi-scale differentiation heterogeneity among cells, genes, and gene modules in osteogenesis. PLOS Genetics, 20(10):e1011319, October 2024. doi:10.1371/journal.pgen.1011319.\">Han <em>et al.</em>, 2024</a>]</span>. This may result from technical bias <span id=\"id17\">[<a class=\"reference internal\" href=\"../references.html#id31\" title=\"Jia Cao, Ling Jin, Zi-Qi Yan, Xiao-Kai Wang, You-You Li, Zun Wang, Yi-Wei Liu, Hong-Ming Li, Zhe Guan, Ze-Hui He, Jiang-Shan Gong, Jiang-Hua Liu, Hao Yin, Yi-Juan Tan, Chun-Gu Hong, Shi-Kai Feng, Yan Zhang, Yi-Yi Wang, Lu-Yue Qi, Chun-Yuan Chen, Zheng-Zhao Liu, Zhen-Xing Wang, and Hui Xie. Reassessing endothelial-to-mesenchymal transition in mouse bone marrow: insights from lineage tracing models. Nature Communications, 14(1):8461, December 2023. doi:10.1038/s41467-023-44312-w.\">Cao <em>et al.</em>, 2023</a>]</span>, such as droplets, although we have performed droplet discovery in <a class=\"reference internal\" href=\"../preprocess/droplet.html\"><span class=\"std std-doc\">droplet discovery</span></a>. It may also be related to epithelial-mesenchymal transition <span id=\"id18\">[<a class=\"reference internal\" href=\"../references.html#id32\" title=\"Jing Yang, Parker Antin, Geert Berx, C\u00e9dric Blanpain, Thomas Brabletz, Marianne Bronner, Kyra Campbell, Amparo Cano, Jordi Casanova, Gerhard Christofori, Shoukat Dedhar, Rik Derynck, Heide L. Ford, Jonas Fuxe, Antonio Garc\u00eda de\u00a0Herreros, Gregory J. Goodall, Anna-Katerina Hadjantonakis, Ruby Y. J. Huang, Chaya Kalcheim, Raghu Kalluri, Yibin Kang, Yeesim Khew-Goodall, Herbert Levine, Jinsong Liu, Gregory D. Longmore, Sendurai A. Mani, Joan Massagu\u00e9, Roberto Mayor, David McClay, Keith E. Mostov, Donald F. Newgreen, M. Angela Nieto, Alain Puisieux, Raymond Runyan, Pierre Savagner, Ben Stanger, Marc P. Stemmler, Yoshiko Takahashi, Masatoshi Takeichi, Eric Theveneau, Jean Paul Thiery, Erik W. Thompson, Robert A. Weinberg, Elizabeth D. Williams, Jianhua Xing, Binhua P. Zhou, and Guojun Sheng. Guidelines and definitions for research on epithelial\u2013mesenchymal transition. Nature Reviews Molecular Cell Biology, 21(6):341\u2013352, June 2020. doi:10.1038/s41580-020-0237-9.\">Yang <em>et al.</em>, 2020</a>]</span>. Due to the uncertainty, we will not discuss this cluster further.</p><p><img alt=\"png\" src=\"../_images/C9-9.png\"/></p>", "a[href=\"#bud-mesenchyme-cap-mesenchyme\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Bud Mesenchyme &amp; Cap Mesenchyme<a class=\"headerlink\" href=\"#bud-mesenchyme-cap-mesenchyme\" title=\"Permalink to this heading\">#</a></h2><p>In the study focused on tooth germ development <span id=\"id1\">[<a class=\"reference internal\" href=\"../references.html#id27\" title=\"Yaofeng Wang, Yifan Zhao, Shubin Chen, Xiaoming Chen, Yanmei Zhang, Hong Chen, Yuansong Liao, Jiashu Zhang, Di Wu, Hongxing Chu, Hongying Huang, Caixia Wu, Shijuan Huang, Huichao Xu, Bei Jia, Jie Liu, Bo Feng, Zhonghan Li, Dajiang Qin, Duanqing Pei, and Jinglei Cai. Single cell atlas of developing mouse dental germs reveals populations of CD24+ and Plac8+ odontogenic cells. Science Bulletin, 67(11):1154\u20131169, June 2022. doi:10.1016/j.scib.2022.03.012.\">Wang <em>et al.</em>, 2022</a>]</span> <span id=\"id2\">[<a class=\"reference internal\" href=\"../references.html#id2\" title=\"Junjun Jing, Jifan Feng, Yuan Yuan, Tingwei Guo, Jie Lei, Fei Pei, Thach-Vu Ho, and Yang Chai. Spatiotemporal single-cell regulatory atlas reveals neural crest lineage diversification and cellular function during tooth morphogenesis. Nature Communications, 13(1):4803, August 2022. doi:10.1038/s41467-022-32490-y.\">Jing <em>et al.</em>, 2022</a>]</span>, the data reveal significant heterogeneity among the datasets at different stages, such as the bud stage and cap stage. We have also observed that the thickening stage mesenchyme (&lt; E13.5) and bud stage (E13.5-E14.5) (See <a class=\"reference internal\" href=\"../preprocess/metadata_process.html\"><span class=\"std std-doc\">Metadata Collection</span></a>) are predominantly present in the C9-1 cluster. The C9-2 cluster is primarily composed of mesenchyme at the cap stage (E14.5-E18.5). Therefore, we assigned the C9-1 cluster as \u201cBud Mesenchyme\u201d and the C9-2 cluster as \u201cCap Mesenchyme\u201d.\n<img alt=\"png\" src=\"../_images/C9_age_bar.png\"/></p><p>Bud mesenchyme highly expressed the genes related to embroyic develpment, such as <a class=\"reference external\" href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=tbx18\">Tbx18</a> and <a class=\"reference external\" href=\"https://www.genecards.org/cgi-bin/carddisp.pl?gene=hic1\">Hic1</a> <span id=\"id3\">[<a class=\"reference internal\" href=\"../references.html#id28\" title=\"Martin Arostegui, R. Wilder Scott, Kerstin B\u00f6se, and T. Michael Underhill. Cellular taxonomy of Hic1+ mesenchymal progenitor derivatives in the limb: from embryo to adult. Nature Communications, 13(1):4989, August 2022. doi:10.1038/s41467-022-32695-1.\">Arostegui <em>et al.</em>, 2022</a>]</span>.\n<img alt=\"png\" src=\"../_images/C9-1-gene.png\"/></p>", "a[href=\"#motivation\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Motivation<a class=\"headerlink\" href=\"#motivation\" title=\"Permalink to this heading\">#</a></h2><p>In level 2 annotation, we devided the mesenchyme into 9 clusters, including four large populations (\u201cC9-1\u201d,\u201dC9-2\u201d,\u201dC9-6\u201d,\u201dC9-7\u201d) and five small populations. In this level, we want to assign the consensus annotation from literature to these clusters.\n<img alt=\"png\" src=\"../_images/anno_C9.png\"/></p><p>Here is a summary of the annotation of each cluster:\n<img alt=\"png\" src=\"../_images/anno_level1_level2.png\"/></p>", "a[href=\"#level-2-annotation\"]": "<h1 class=\"tippy-header\" style=\"margin-top: 0;\">Level 2 annotation<a class=\"headerlink\" href=\"#level-2-annotation\" title=\"Permalink to this heading\">#</a></h1><h2>Motivation<a class=\"headerlink\" href=\"#motivation\" title=\"Permalink to this heading\">#</a></h2><p>In level 2 annotation, we devided the mesenchyme into 9 clusters, including four large populations (\u201cC9-1\u201d,\u201dC9-2\u201d,\u201dC9-6\u201d,\u201dC9-7\u201d) and five small populations. In this level, we want to assign the consensus annotation from literature to these clusters.\n<img alt=\"png\" src=\"../_images/anno_C9.png\"/></p><p>Here is a summary of the annotation of each cluster:\n<img alt=\"png\" src=\"../_images/anno_level1_level2.png\"/></p>", "a[href=\"#stress-related\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Stress-related<a class=\"headerlink\" href=\"#stress-related\" title=\"Permalink to this heading\">#</a></h2><p>The cluster (C9-4) highly expresses genes related to stress. GO enrichment analysis shows enrichment in <code class=\"docutils literal notranslate\"><span class=\"pre\">Regulation</span> <span class=\"pre\">Of</span> <span class=\"pre\">Angiogenesis</span></code>, <code class=\"docutils literal notranslate\"><span class=\"pre\">Vasculature</span> <span class=\"pre\">Development</span></code>, and <code class=\"docutils literal notranslate\"><span class=\"pre\">Nitric</span> <span class=\"pre\">Oxide</span> <span class=\"pre\">Metabolic</span> <span class=\"pre\">Process</span></code>. Most importantly, this cluster is only observed in one dataset <span id=\"id19\">[<a class=\"reference internal\" href=\"../references.html#id9\" title=\"S.Y. Luo, S. Wang, Z.X. Liu, Q. Bian, and X.D. Wang. Six1 Regulates Mouse Incisor Development by Promoting Dlx1/2/5 Expression. Journal of Dental Research, 103(10):1017\u20131027, September 2024. doi:10.1177/00220345241256286.\">Luo <em>et al.</em>, 2024</a>]</span>. Therefore, we believe this cluster is suspicious.\n<img alt=\"png\" src=\"../_images/C9-GO.png\"/>\n<img alt=\"png\" src=\"../_images/umap_project.png\"/></p>", "a[href=\"#odontoblast\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Odontoblast<a class=\"headerlink\" href=\"#odontoblast\" title=\"Permalink to this heading\">#</a></h2><p>Base on the expression of Phex <span id=\"id13\">[<a class=\"reference internal\" href=\"../references.html#id29\" title=\"Florian Hermans, Celine Bueds, Lara Hemeryck, Ivo Lambrichts, Annelies Bronckaers, and Hugo Vankelecom. Establishment of inclusive single-cell transcriptome atlases from mouse and human tooth as powerful resource for dental research. Frontiers in Cell and Developmental Biology, 2022.\">Hermans <em>et al.</em>, 2022</a>]</span>, Smpd3 <span id=\"id14\">[<a class=\"reference internal\" href=\"../references.html#id30\" title=\"Yingchun Zheng, Ting Lu, Leitao Zhang, Zhongzhi Gan, Aoxi Li, Chuandong He, Fei He, Sha He, Jian Zhang, and Fu Xiong. Single-cell RNA-seq analysis of rat molars reveals cell identity and driver genes associated with dental mesenchymal cell differentiation. BMC Biology, 22(1):1\u201313, September 2024. doi:10.1186/s12915-024-01996-w.\">Zheng <em>et al.</em>, 2024</a>]</span>, we can easily identify the odontoblast cluster C9-8.\n<img alt=\"png\" src=\"../_images/C9-8.png\"/></p>", "a[href=\"#apical-pulp-distal-pulp\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Apical pulp &amp; Distal pulp<a class=\"headerlink\" href=\"#apical-pulp-distal-pulp\" title=\"Permalink to this heading\">#</a></h2><p>Postnatal mesenchyme in <a class=\"reference internal\" href=\"Level1_anno.html\"><span class=\"std std-doc\">level 1</span></a> annotation can be devided into two main clusters: C9-6 and C9-7. The most important factors that drive the heterogeneity of postnatal pulp may be the spatial location. Apical pulp and distal pulp are have different gene expression profiles <span id=\"id5\">[<a class=\"reference internal\" href=\"../references.html#id29\" title=\"Florian Hermans, Celine Bueds, Lara Hemeryck, Ivo Lambrichts, Annelies Bronckaers, and Hugo Vankelecom. Establishment of inclusive single-cell transcriptome atlases from mouse and human tooth as powerful resource for dental research. Frontiers in Cell and Developmental Biology, 2022.\">Hermans <em>et al.</em>, 2022</a>]</span> <span id=\"id6\">[<a class=\"reference internal\" href=\"../references.html#id30\" title=\"Yingchun Zheng, Ting Lu, Leitao Zhang, Zhongzhi Gan, Aoxi Li, Chuandong He, Fei He, Sha He, Jian Zhang, and Fu Xiong. Single-cell RNA-seq analysis of rat molars reveals cell identity and driver genes associated with dental mesenchymal cell differentiation. BMC Biology, 22(1):1\u201313, September 2024. doi:10.1186/s12915-024-01996-w.\">Zheng <em>et al.</em>, 2024</a>]</span>.\nWe checked the expression of the marker genes of apical and distal pulp in these two clusters, which align with our expectation.</p><p><strong>Apical Pulp</strong></p>", "a[href=\"#chondrocyte\"]": "<h2 class=\"tippy-header\" style=\"margin-top: 0;\">Chondrocyte<a class=\"headerlink\" href=\"#chondrocyte\" title=\"Permalink to this heading\">#</a></h2><p>We observed a cluster (C9-3) that highly expresses chondrogenic marker genes, such as Col2a1, Matn3, and Sox5 <span id=\"id15\">[<a class=\"reference internal\" href=\"../references.html#id19\" title=\"Litian Han, Yaoting Ji, Yiqian Yu, Yueqi Ni, Hao Zeng, Xiaoxin Zhang, Huan Liu, and Yufeng Zhang. Trajectory-centric framework TrajAtlas reveals multi-scale differentiation heterogeneity among cells, genes, and gene modules in osteogenesis. PLOS Genetics, 20(10):e1011319, October 2024. doi:10.1371/journal.pgen.1011319.\">Han <em>et al.</em>, 2024</a>]</span>. Chondrocytes are generally considered unrelated to odontogenesis, but considering that some groups take samples from the embryonic head, this is reasonable in our atlas.\n<img alt=\"png\" src=\"../_images/C9-3.png\"/></p>"}
skip_classes = ["headerlink", "sd-stretched-link"]

window.onload = function () {
    for (const [select, tip_html] of Object.entries(selector_to_html)) {
        const links = document.querySelectorAll(`div.content ${select}`);
        for (const link of links) {
            if (skip_classes.some(c => link.classList.contains(c))) {
                continue;
            }

            tippy(link, {
                content: tip_html,
                allowHTML: true,
                arrow: true,
                placement: 'auto-start', maxWidth: 500, interactive: false,
                onShow(instance) {MathJax.typesetPromise([instance.popper]).then(() => {});},
            });
        };
    };
    console.log("tippy tips loaded!");
};
